newIDE/app/src/UI/MaterialUISpecificUtil.js

```diff
@@ -6,3 +6,46 @@ export const isNoDialogOpened = (): boolean => {
     'body > div[role="presentation"].MuiDialog-root'
   );
 };
+
+export const isElementADialog = (
+  element: Element,
+  options?: { isVisible: true }
+) => {
+  const isDialog =
+    element.tagName === 'DIV' &&
+    element.getAttribute('role') === 'presentation';
+  if (!isDialog) return false;
+  if (options && options.isVisible) {
+    return !element.getAttribute('aria-hidden');
+  }
+  return true;
+};
+
+/**
+ * Checks if an input element is a Material UI checkbox.
+ */
+export const isMuiCheckbox = (element: HTMLElement): boolean => {
+  const typeAttribute = element.attributes.getNamedItem('type');
+  return (
+    typeAttribute &&
+    typeAttribute.value === 'checkbox' &&
+    'indeterminate' in element.dataset
+  );
+};
+
+/**
+ * Returns the value of a Material UI checkbox.
+ * Material UI does not use the value of the input element
+ * (See https://v4.mui.com/api/checkbox/#props) so we have to check
+ * the classes of the grand parent of the input element.
+ */
+export const getMuiCheckboxValue = (element: HTMLElement): boolean => {
+  const { parentElement } = element;
+  if (parentElement) {
+    const { parentElement: grandParentElement } = parentElement;
+    return grandParentElement
+      ? grandParentElement.classList.contains('Mui-checked')
+      : false;
+  }
+  return true;
```

Comment:
feels weird returning true if the element is not found no?

Reply:
You're right, I'll change it